src/test/java/com/artipie/asto/S3StorageTest.java

```diff
@@ -71,6 +79,34 @@ void shouldUploadObjectWhenSave(final AmazonS3 client) throws Exception {
         MatcherAssert.assertThat(downloaded, Matchers.equalTo(data));
     }
 
+    @Test
+    void shouldFailToSaveMultipartUploadWhenFailedToReadContent(final AmazonS3 client) {
+        final String bucket = UUID.randomUUID().toString();
+        client.createBucket(bucket);
+        final String key = "fail";
+        final CompletableFuture<Void> saved = this.storage(bucket).save(
```

Comment:
@olegmoz Let's inline `saved`

Reply:
@HDouss done here and in other test `shouldAbortMultipartUploadWhenFailedToReadContent`